convert2rhel/logger.py

```diff
@@ -81,52 +79,47 @@ class LogLevelTask(object):
     label = "TASK"
 
 
-class LogLevelFile(object):
-    level = 5
-    # Label messages DEBUG as it is contains the same messages as debug, just that they always go
-    # to the log file.
-    label = "DEBUG"
-
-
 def setup_logger_handler(log_name, log_dir):
     """Setup custom logging levels, handlers, and so on. Call this method
     from your application's main start point.
         log_name = the name for the log file
         log_dir = path to the dir where log file will be presented
     """
-    # set custom labels
+    # set the TASK custom label
     logging.addLevelName(LogLevelTask.level, LogLevelTask.label)
-    logging.addLevelName(LogLevelFile.level, LogLevelFile.label)
     logging.Logger.task = _task
-    logging.Logger.file = _file
-    logging.Logger.debug = _debug
     logging.Logger.critical = _critical
 
     # enable raising exceptions
     logging.raiseExceptions = True
-    # get root logger
+    # get the highest level app logger
     logger = logging.getLogger("convert2rhel")
-    # propagate
+    # propagate log messages up to the root logger to be able to capture them in unit tests
+    # refence: https://github.com/oamg/convert2rhel/pull/179
     logger.propagate = True
-    # set default logging level
-    logger.setLevel(LogLevelFile.level)
+    # set the DEBUG level as the lowest allowed level to be handled by convert2rhel
+    logger.setLevel(logging.DEBUG)
 
     # create sys.stdout handler for info/debug
     stdout_handler = logging.StreamHandler(sys.stdout)
     formatter = CustomFormatter("%(message)s")
     formatter.disable_colors(should_disable_color_output())
     stdout_handler.setFormatter(formatter)
+    debug_flag_filter = DebugFlagFilter()
+    stdout_handler.addFilter(debug_flag_filter)
+    # Set the DEBUG level as the lowest allowed level to be printed to stdout.
+    # Whether a debug message is actually printed or not is decided in DebugFlagFilter.
     stdout_handler.setLevel(logging.DEBUG)
```

Comment:
Does it not work to set this to `logging.DEBUG` if `--debug` is given, otherwise set it to `loggin.INFO`?  (If the problem is a circular dependency [Need toolopts to know if `--debug` was given but toolopts requires that logging has been setup in case it needs to log its values, the way that I've setup logging in other projects is to have two phases.  Phase 1 sets it up with default values.  Then parse the command line, read the config file, consult environment variables, etc.  Then Phase 2, reconfigure portions of logging that user settings can affect.

That's a bigger change than this, but I think it's better structure.  And at the risk of premature optimization, it's also less work for the computer.  Adjusting the log level results in less work being done if the log message is at a less important level.  Setting a filter results in more work being done for every log message emitted.